locking/locks.go

```diff
@@ -47,15 +50,20 @@ func NewClient(cfg *config.Configuration) (*Client, error) {
 	return &Client{cfg, apiClient, store}, nil
 }
 
+// Close this client instance; must be called to dispose of resources
+func (c *Client) Close() error {
+	return c.cache.Save()
+}
+
 // LockFile attempts to lock a file on the current remote
 // path must be relative to the root of the repository
 // Returns the lock id if successful, or an error
-func (c *Client) LockFile(path string) (id string, e error) {
+func (c *Client) LockFile(path string) (Lock, error) {
```

Comment:
Actually I remember this was deliberately by value now. Returning a non-pointer is important because I don't want to allow the client to modify the original lock, because it's cached and therefore they could corrupt the cache by doing that. This outweighs the zero value Lock concern IMO. Yes I could return a pointer and copy it but it's unnecessary complexity. A zero value in the event of an error is idiomatic anyway (nil is just a zero value pointer after all).